pkg/controller/psbackup/controller_test.go

```diff
@@ -0,0 +1,180 @@
+package psbackup
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/platform"
+)
+
+func TestBackupStatusErrStateDesc(t *testing.T) {
+	namespace := "some-namespace"
+	clusterName := "some-name"
+	backupName := "backup1"
+	q, err := resource.ParseQuantity("1Gi")
+	if err != nil {
+		t.Fatal(err)
+	}
+	storageName := "some-storage"
+
+	tests := []struct {
+		name      string
+		cluster   *apiv1alpha1.PerconaServerMySQL
+		cr        *apiv1alpha1.PerconaServerMySQLBackup
+		stateDesc string
+	}{
+		{
+			name: "Test without cluster",
+			cr: &apiv1alpha1.PerconaServerMySQLBackup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      backupName,
+					Namespace: namespace,
+				},
+				Spec: apiv1alpha1.PerconaServerMySQLBackupSpec{
+					ClusterName: clusterName,
+					StorageName: storageName,
+				},
+			},
+			stateDesc: fmt.Sprintf("PerconaServerMySQL %s in namespace %s is not found", clusterName, namespace),
+		},
+		{
+			name: "Test without enabled backup section",
+			cr: &apiv1alpha1.PerconaServerMySQLBackup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      backupName,
+					Namespace: namespace,
+				},
+				Spec: apiv1alpha1.PerconaServerMySQLBackupSpec{
+					ClusterName: clusterName,
+					StorageName: storageName,
+				},
+			},
+			cluster: &apiv1alpha1.PerconaServerMySQL{
```

Comment:
I tried to improve this test. Please take a look and tell me what do you think about this: https://github.com/percona/percona-server-mysql-operator/pull/261/commits/d77668c65e189be01f6be50bcb3b9fa690cb00fa

Reply:
I generally like it, like the use of generics, but maybe for this case in particular it maybe is a bit too sophisticated. With gingko I think we can make it a bit simpler.

Every controller has (will have) its own `suite_test.go` file, like `pkg/controller/ps/suite_test.go` and that file can contain some function that will read and load default CR and it will be available for every test. Yes, we will repeat some things (for three controllers), but I don't think it is that bad and I always favor a bit of duplication for more simplicity. Also this way we will don't need this new `testutil` package.

Again, I like your solution, clever and nice use of generics, but as I said, maybe we can have it simpler.

@egegunes give us your 2 cents.

Reply:
I guess we can resolve this one @pooknull , after you updated the branch, you have a way of loading the default CR.

Reply:
I have removed the `testutil` package https://github.com/percona/percona-server-mysql-operator/pull/261/commits/58d486f13fee27099617c85ff4fab382e8947958